### PR TITLE
Ref#4596 Cover different db types for jdbc

### LINKED ISSUE #4596: Expand JDBC test coverage
The following scenarios from the [documentation](https://camel.apache.org/components/3.20.x/jdbc-component.html) are not covered in Camel Quarkus:

1. [Generated keys](https://camel.apache.org/components/3.20.x/jdbc-component.html#_generated_keys)
2. [Using named parameters](https://camel.apache.org/components/3.20.x/jdbc-component.html#_using_named_parameters)
3. Polling databes every time interval - part of the [samples](https://camel.apache.org/components/3.20.x/jdbc-component.html#_sample_polling_the_database_every_minute).
4. Move data between datasources - part of the [samples](https://camel.apache.org/components/3.20.x/jdbc-component.html#_sample_move_data_between_data_sources).
 
Because JDBC extension leverages quarkus extension (see the [documentaion](https://quarkus.io/guides/datasource)) it would be nice to cover major features - it is optional.

1.  Cover different db types for jdbc (based on the [doc](https://quarkus.io/guides/datasource#jdbc-datasource-2)) - DB2, Derby, H2, MariaDB, Microsoft SQL Server, MySQL, Oracle and PostgreSQL.

### PR DESCRIPTION
**Pull request by ldrozdo**
Fixes #4596

Last part of expanding JDBC integration tests: covering different db types.

I've disabled some tests because of:
1. #4750 for postgresql 
2. [For derby](https://github.com/quarkusio/quarkus/issues/23083)

I didn't use quarkus-flyway yet, as we discussed in #4730, it can be covered in next PRs.

### CODE REVIEW

**Comment by ldrozdo** [comment]
#4750 was resolved, I've fixed it in the tests.

File: integration-tests/jdbc/src/main/java/org/apache/camel/quarkus/component/jdbc/CamelResource.java [hunk 1]
```diff
@@ -88,22 +87,22 @@ public String getSpeciesById(@PathParam("id") String id) throws Exception {
...
-                .requestBody("jdbc:camel-ds?outputType=SelectList", "select * from camels where id = " + id, List.class);
+                .requestBody("jdbc:cameldb?outputType=SelectList", "select * from camels where id = " + id, List.class);
 
         if (result.isEmpty()) {
             throw new IllegalStateException("Expected at least 1 camel result but none were found");
         }
 
         LinkedHashMap<String, Object> data = result.get(0);
-        return data.get("SPECIES") + " " + data.get("ID");
+        return data.toString();
```
Comment (aldettinger) [inline integration-tests/jdbc/src/main/java/org/apache/camel/quarkus/component/jdbc/CamelResource.java:97]: Is it really better to use toString() ? What happen if the toString() implementation changes ?

File: integration-tests/jdbc/src/main/java/org/apache/camel/quarkus/component/jdbc/CamelResource.java [hunk 2]
```diff
@@ -183,4 +183,47 @@ public String moveBetweenDatasources() throws Exception {
...
+        try (Connection conn = dataSource.getConnection()) {
+            try (Statement statement = conn.createStatement()) {
+                try (InputStream is = Thread.currentThread().getContextClassLoader()
+                        .getResourceAsStream("sql/" + fileName);
+                        InputStreamReader isr = new InputStreamReader(is);
+                        BufferedReader reader = new BufferedReader(isr)) {
+
+                    reader.lines().filter(s -> s != null && !"".equals(s) && !s.startsWith("--")).forEach(s -> {
+                        try {
+                            statement.execute(s);
```
Comment (aldettinger) [inline integration-tests/jdbc/src/main/java/org/apache/camel/quarkus/component/jdbc/CamelResource.java:196]: Ok, it's executing each line as a separate statement.

File: integration-tests/jdbc/src/main/resources/application.properties [hunk 3]
```diff
@@ -18,6 +18,11 @@
 #
 # Quarkus :: DS
 #
-quarkus.datasource.camel-ds.jdbc.url=jdbc:h2:tcp://localhost/mem:test
-quarkus.datasource.camel-ds.db-kind=h2
-quarkus.datasource.camel-ds.jdbc.max-size=8
+#quarkus.datasource.camel-ds.jdbc.url=jdbc:h2:tcp://localhost/mem:test
+#quarkus.datasource.camel-ds.db-kind=h2
+#quarkus.datasource.camel-ds.jdbc.max-size=8
```
Comment (aldettinger) [inline integration-tests/jdbc/src/main/resources/application.properties:23]: If this is like default value, could we remove the commented lines completely ?

File: integration-tests/jdbc/src/main/java/org/apache/camel/quarkus/component/jdbc/CamelResource.java (hunk 2 above)
Comment (aldettinger) [reply integration-tests/jdbc/src/main/java/org/apache/camel/quarkus/component/jdbc/CamelResource.java:196]: Then, maybe we could put a comment ?

**Comment by aldettinger** [comment]
Well done @ldrozdo . That's good addition. There are few questions in the review.

> It seems that we can introduce a [group testing](https://camel.apache.org/camel-quarkus/3.0.x/contributor-guide/extension-testing.html#_grouping) for all the supported db.

Interesting idea @zhfeng , maybe it could be another ticket ?

**Comment by ldrozdo** [comment]
Thanks @aldettinger for your comments, I implemented them. Should I create issue for group testing which @zhfeng suggested?

**Comment by zhfeng** [comment]
@ldrozdo yeah, please create a follow up issue for group testing. Thanks!

**Comment by ldrozdo** [comment]
@aldettinger I'm sorry, the checks are failing and I can't figure out why. Everything works on my side. Can you please have a look, maybe you will spot the issue? Thank you very much.

**Comment by aldettinger** [comment]
> @aldettinger I'm sorry, the checks are failing and I can't figure out why. Everything works on my side. Can you please have a look, maybe you will spot the issue? Thank you very much.

@ldrozdo The previous failure was just a transient connect failure we have more often those days. I think @essobedo was really nice to relaunch the job :+1:  A bit of fingers crossed and it should pass :)

**Review by zhfeng — APPROVED** [review]

**Pull request merged**